### DISCUSSION TITLE
Profiles for different workplace setups

### DISCUSSION THREAD

**Discussion by Speed1** [nikitabobko/AeroSpace#712] [questions-and-answers]
Hi,

I have a question about creating two different workplace profiles in aerospace.

In the office, I use two 27” external displays, while at home, I have a single 37” external display. Unfortunately, my “on-windows-detect” settings don’t work seamlessly across both setups.

I considered creating two different modes and assigning specific “on-windows-detect” settings to each, but it seems there’s no “if” condition available for [this mode](https://nikitabobko.github.io/AeroSpace/guide#on-window-detected-callback).

The only workaround I could think of is creating two separate configuration files and loading them based on the location.

Is there a more elegant solution to achieve this?

Thanks

**Comment by fgeck** [comment]
I have the exact same problem. 
Basically I would like to have different profiles based on my external monitor setup.
Depending on the external monitors I assign the workspaces to them and arrange the apps in the workspaces.
Especially the app arrangement is completely different when I use Macbook only vs having one 32" monitor connected, or Macbook only vs having 2 x 27" monitors connected.

The only thing I can think of - given the current features available -is having different aerospace config files, and using a symlink to the currently active config (aka. profile).
To make switching more convenient one could build a small bash script and include this in raycast (see [raycast script commands](https://www.raycast.com/blog/getting-started-with-script-commands)).
But this is just a workaround which I do not prefer.
